packages/google_sign_in/example/lib/main.dart

```diff
@@ -94,18 +97,45 @@ class SignInDemoState extends State<SignInDemo> {
     return null;
   }
 
-  Future<Null> _handleSignIn() async {
-    try {
-      await _googleSignIn.signIn();
-    } catch (error) {
-      print(error);
-    }
+  void _handleSignIn() {
+    _handleGoogleSignIn()
+        .then((GoogleSignInAccount user) => _afterSignIn)
+        .catchError((dynamic error) => _onSignInError);
   }
 
-  Future<Null> _handleSignOut() async {
+  Future<GoogleSignInAccount> _handleGoogleSignIn() async {
+    await _googleSignIn.signIn();
+    return _googleSignIn.currentUser;
+  }
+
+  Future<Null> _handleGoogleSignOut() async {
     _googleSignIn.disconnect();
   }
 
+  void _afterSignIn(GoogleSignInAccount user) {
+    _scaffoldKey.currentState.showSnackBar(new SnackBar(
+        content: new Text("You're signed in " + user.displayName)),
+    );
+  }
+
+  void _onSignInError(dynamic error) {
+    print(error);
+
+    if (error is PlatformException &&
+        error.details.toString().contains('canceled')) {
+      print("User canceled sign in");
```

Comment:
Correct. It should not do this. It would be great if we fixed it. It should return null as the signin account if the user cancelled. I would be happy to review the PR if you send one my way :-)